app/components/properties-panel/properties-panel.component.html

```diff
@@ -1,77 +1,41 @@
 <div id="wrapper">
     <h2>Properties</h2>
-    <div *ngIf="isEmpty()" id="empty">
+    <div *ngIf="isEmpty" id="empty">
         <p>empty</p>
     </div>
 
-    <div *ngIf="!isEmpty()">
-        <table *ngFor="let group of groups" class="group-table">
-            <template [ngIf]="selectedEntity[group[1]] && selectedEntity[group[1]].properties.length > 0">
-                <tr><th colspan="2">{{group[0]}}</th></tr>
-                <tr *ngFor="let pair of selectedEntity[group[1]].properties">
-                    <td class="label">{{pair[0]}}</td>
-                    <td [ngSwitch]="pair[1].kind">
-                        <div *ngSwitchCase="'Number'">
-                            <input [ngModel]="selectedEntity[group[1]].get(pair[0])" (ngModelChange)="doChange(selectedEntity, group[1], pair[0], $event)" type="number" 
-                                class="number-input"/>
+    <div *ngIf="!isEmpty">
+        <table *ngFor="let name of fieldNames" class="group-table">
+            <template [ngIf]="fields[name]">
+                <tr><th colspan="2">{{name}}</th></tr>
+                <tr *ngFor="let triple of fields[name]">
```

Comment:
Unless you have a better idea, @Sheyne, I'm going to make a function that wraps up triple in a object with named properties (ie `let triple of wrap(fields[name])` -> `triple.name`, `triple.path`, etc...) for readabilities sake.